### ISSUE TITLE
Monadic productions with threaded lexer produces ill-formed haskell code

### ISSUE THREAD

**Issue by jgallag88** [haskell/happy#35]
I am using Alex along with Happy, and I have a rule that looks something like:

```
gdecl   :  typedef type ident ';'         {%^ mkTypDef $2 $3 }
```

The documentation says that this is the proper way to get a hold of the lookahead token when using a threaded lexer (%lexer). Running Happy on this produces a .hs file which GHC can't parse:

```
Parser.hs:301:47: parse error on input ‘tk’
```

Here is the function that contains the error:

```
happyReduce_6 = happyMonadReduce 4# 2# happyReduction_6
happyReduction_6 (happy_x_4 `HappyStk`
        happy_x_3 `HappyStk`
        happy_x_2 `HappyStk`
        happy_x_1 `HappyStk`
        happyRest) tk 
         = happyThen (case happyOut21 happy_x_2 of { happy_var_2 ->
        case happyOutTok happy_x_3 of { (LocTok _ (TokIdent happy_var_3)) ->
        ( mkTypDef happy_var_2 happy_var_3)}} tk        <-- Error Here
        ) (\r -> happyReturn (happyIn6 r))
```

Normal monadic productions, i.e. {% expn}, work fine. I am using Happy 1.19.4 and GHC 7.8.3. I can't post the entire Happy file, but if necessary I could probably put together a minimal working example.

**Comment by simonmar** [comment]
Any chance you could turn the example into a test case?

**Comment by pgarrison** [comment]
Yeah, I'll work on that.

**Comment by harpocrates** [comment]
I just ran into this exact bug (and the merged PR solves it for me). I see that the last release of Happy (1.19.5) on Hackage was over a year ago. When is 1.19.6 scheduled to be released?

**Comment by sgraf812** [comment]
FTR, the fix (#54) was that we need to wrap parens around the `case ... of {}` in `case ... of {} tk`.

Let's just close this without a reproducer; at the moment it just steals bandwidth.

**Comment by pgarrison** [comment]
What an old thread -- I believe this PR was my first ever contribution to an open source project! Somehow my plan to make a test case got lost in the last 9 years 😂
